onnxruntime/core/framework/cloud_invoker.cc

```diff
@@ -0,0 +1,290 @@
+    options.client_timeout_ = 0;
+
+    err = triton_client_->Infer(&results, options, triton_inputs, triton_outputs,
+                                http_headers, tc::Parameters(),
+                                tc::InferenceServerHttpClient::CompressionType::NONE,  //to-do: support compression in config?
+                                tc::InferenceServerHttpClient::CompressionType::NONE);
+    CHECK_TRITON_ERR(err, "Triton client failed to do inference");
+
+    if (ort_outputs.empty()) {
+      ort_outputs.resize(output_names.size());
+    }
+
+    int output_index = 0;
+    std::unique_ptr<tc::InferResult> results_ptr;
+    results_ptr.reset(results);
+    iter = output_names.begin();
+
+    while (iter != output_names.end()) {
+      std::vector<int64_t> dims;
+      err = results_ptr->Shape(*iter, &dims);
+      CHECK_TRITON_ERR(err, (std::string{"Failed to get shape for output "} + *iter).c_str());
+
+      std::string type;
+      err = results_ptr->Datatype(*iter, &type);
+      CHECK_TRITON_ERR(err, (std::string{"Failed to get type for output "} + *iter).c_str());
+
+      const uint8_t* raw_data = {};
+      size_t raw_size;
+      err = results_ptr->RawData(*iter, &raw_data, &raw_size);
+      CHECK_TRITON_ERR(err, (std::string{"Failed to get raw data for output "} + *iter).c_str());
+
+      auto output_tensor = CreateTensor(type, dims);
+      if (!output_tensor) {
+        return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Failed to create output tensor for output", *iter);
+      }
+      //todo - can we skip memcpy?
+      memcpy(output_tensor->MutableDataRaw(), raw_data, raw_size);
+      ort_outputs[output_index].Init(output_tensor.release(), tensor_type, tensor_type->GetDeleteFunc());
+
+      ++iter;
+      ++output_index;
+    }
+  } catch (const std::exception& ex) {
+    return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, "Caught exception in TritonInvokder::Send", ex.what());
+  }
+  return Status::OK();
+}
+
+bool CloudEndPointInvoker::ReadConfig(const char* config_name, bool& config_val, bool required) {
+  if (config_.count(config_name)) {
+    config_val = config_[config_name] == "true" || config_[config_name] == "True" || config_[config_name] == "TRUE" || config_[config_name] == "1";
+  } else if (required) {
+    status_ = ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Triton invoker failed to initialize due to missed config: ", config_name);
+  }
+  return status_.IsOK();
+}
+
+bool CloudEndPointInvoker::ReadConfig(const char* config_name, std::string& config_val, bool required) {
+  if (config_.count(config_name)) {
+    config_val = config_[config_name];
+  } else if (required) {
+    status_ = ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT, "Triton invoker failed to initialize due to missed config: ", config_name);
+  }
+  return status_.IsOK();
+}
+
+std::unique_ptr<CloudEndPointInvoker> CloudEndPointInvoker::CreateInvoker(const CloudEndPointConfig& config) {
+  if (config.count(kCloudEndpointType)) {
```

Comment:
Can use find and the returned iterator instead of looking up twice.

Reply:
Switch to find.